tasks/oss.yml

```diff
@@ -3,58 +3,51 @@
   package:
     name: beegfs-storage
     state: present
-  become: True
+  become: yes
+
+- name: Create a list of devices that are mounted
+  set_fact:
+    mounted_path_dev_items: "{{ ansible_mounts | map(attribute='device') | list }}"
 
 - name: Ensures BeeGFS storage service is not running
   service:
-    name: "beegfs-storage"
-    state: "stopped"
-  become: True
+    name: beegfs-storage
+    state: stopped
+  become: yes
 
 - name: Prepare storage devices
   block:
-  - name: Ensure storage device is not currently mounted
-    mount:
-      path: "{{ item }}"
-      state: unmounted
-    with_items:
-    - "{{ beegfs_path_oss }}"
-  - name: Format filesystem
+  - name: Attempt to format if the device is not mounted
+    vars:
+      path_dev_item: "{{ beegfs_path_dev }}/{{ item }}"
     filesystem:
-      dev: "{{ item }}"
+      dev: "{{ path_dev_item }}"
       fstype: "{{ beegfs_fstype }}"
-      force: "{{ beegfs_force_format }}"
```

Comment:
The force has been removed here, what if the disk already has a file system on it?

Reply:
Then the user ought to intervene no?

Reply:
Well yes, but removing the force option has removed a feature of the role, where if you set `beegfs_force_format` to true then it will deploy a file system on disks even if they already contain data. Another option would be to provide a mechanism for wiping devices prior to creating the cluster.

Reply:
Its been reintroduced.